pango/src/functions.rs

```diff
@@ -93,3 +93,65 @@ pub fn extents_to_pixels(
         ffi::pango_extents_to_pixels(inclusive.to_glib_none_mut().0, nearest.to_glib_none_mut().0);
     }
 }
+
+#[doc(alias = "pango_itemize")]
+pub fn itemize(
+    context: &Context,
+    text: &str,
+    start_index: i32,
+    length: i32,
+    attrs: &AttrList,
+    cached_iter: Option<&AttrIterator>,
+) -> Vec<Item> {
+    let total_length = text.len() as i32;
+    assert!(
+        start_index >= 0 && start_index < total_length,
+        "start_index is out of range"
+    );
+    assert!(
+        length >= 0 && start_index.checked_add(length).unwrap() < total_length,
```

Comment:
This is wrong unfortunately, sorry for missing that. Needs to be `<= total_length`.

Reply:
https://github.com/gtk-rs/gtk-rs-core/pull/1816